### PR TITLE
Simplify html generator & Add line numbers

### PR DESCRIPTION
**Pull request by bogdancondorachi**
Hello @ryangjchandler,

First, I want to express my gratitude for bringing such a difficult project to life! :)

I see that line numbers are currently supported through the `data-line` attribute, which is then enabled with CSS to display the numbers in the generated HTML. However, what if we generate the line numbers directly in the HTML? By using an inline `user-select: none` style to prevent selection and adding a `line-number` class so we enable further customization while simplifying the structure.

Further, we need a way to enable (should be disabled by default) line numbers but I'll let you decide on how would you do it. Here's an idea..

```php
$html = $phiki->codeToHtml(
    <<<'PHP'
    echo "Hello, world!";
    PHP,
    Grammar::Php,
    Theme::GithubDark,
)->withLineNumbers(startAt: 5); // also nice to have option for line number start
``` 

**Output:**
```html
<span class="line">
  <span class="line-number" style="-webkit-user-select: none; user-select: none;"> 5</span>
  <span class="token" style="color: #79b8ff;">echo</span>
  <span class="token"> </span>
  <span class="token" style="color: #9ecbff;">"</span>
  <span class="token" style="color: #9ecbff;">Hello, world!</span>
  <span class="token" style="color: #9ecbff;">"</span>
  <span class="token">;</span>
  <span class="token"> </span>
 </span>
```

**Customization:**
```css
pre code .line-number {
    margin-right: 1rem;
    /* Suggestion: would be also nice to add the theme base color as inline style with an hex alpha value */
    color: #7683904D; /*(4D = 30% transperency)*/
    text-align: right;
}
``` 

Let me know your thoughts on this approach!

### CODE REVIEW

**Comment by ryangjchandler** [comment]
I think this would be a good edition – I'm going to be refactoring the HTML generation slightly so that it's less clunky, the string concatenation is quite annoying and hard to follow 😅 

I also want to slightly refine the generation stuff so that you get a `Stringable` object back which would make things like `->withLineNumbers()` and other modifiers really easy to add.

**Comment by bogdancondorachi** [comment]
Sounds like you already have a plan and I could not agree more on the HTML generator 😅. Maybe have a look at [Tempest Highlight](https://github.com/tempestphp/highlight) as I think he's done a good job on it's structure.

As of the modifier name you could also opt in for `withGutter(startAt: 10)`

Also for markdown would be nice to be possible to mention starting line in order to change it

```
``php{10}
echo "Hello, world!";
``
```

**Comment by bogdancondorachi** [comment]
@ryangjchandler I've made some adjustments so this can be merged until further refactoring.

- added a `withGutter` parameter that is false by default
- added `editorLineNumber.foreground` color to the line-number style

Edit:
- added a null check for themes that don't support `editorLineNumber.foreground`
- compose the line-number styles as array so will not include `color` if it returns null

Also the `withWrapper` been moved as the fifth parameter based on its importance so the readme would need to be updated.

Let me know your thoughts on this approach!

**Comment by bogdancondorachi** [comment]
> I think this would be a good edition – I'm going to be refactoring the HTML generation slightly so that it's less clunky, the string concatenation is quite annoying and hard to follow 😅
> 
> I also want to slightly refine the generation stuff so that you get a `Stringable` object back which would make things like `->withLineNumbers()` and other modifiers really easy to add.

@ryangjchandler  I took some time to refactor the HTML generator to streamline the build process and improve it's readability. The build process is now simpler and easier to follow. Let me know when you got some time to review this :)

**Comment by ryangjchandler** [comment]
Thank you @bogdancondorachi, I've merged this in as is and I'll continue on with your work.

Sorry for the delay in getting back to you, I really appreciate the time you put into this!
